doc/source/whatsnew/v0.25.0.rst

```diff
@@ -237,6 +237,7 @@ Groupby/Resample/Rolling
 - Bug in :meth:`pandas.core.groupby.DataFrameGroupBy.nunique` in which the names of column levels were lost (:issue:`23222`)
 - Bug in :func:`pandas.core.groupby.GroupBy.agg` when applying a aggregation function to timezone aware data (:issue:`23683`)
 - Bug in :func:`pandas.core.groupby.GroupBy.first` and :func:`pandas.core.groupby.GroupBy.last` where timezone information would be dropped (:issue:`21603`)
+- Bug in :func:`Series.groupby` where using ``groupby`` with a :class:`MultiIndex` Series with a list of labels equal to the length of the series caused incorrect grouping (:issue:`25704`)
```

Comment:
I think this is a little hard to read but potential incorrect anyway. Are you sure this stipulation about the length of the supplied labels w.r.t to the length of the series itself is a requirement to reproduce the issue? 

I did the following locally and still saw the problem:

```python
In [24]:     index_array = [
    ...:         ['x', 'x'],
    ...:         ['a', 'b'],
    ...:         ['k', 'k'],
    ...:         ['j', 'j'],
    ...:     ]
    ...:     index_names = ['first', 'second', 'third', 'fourth']
    ...:     ri = pd.MultiIndex.from_arrays(index_array, names=index_names)
    ...:     s = pd.Series(data=[1, 2], index=ri)
    ...:     result = s.groupby(['first', 'third']).sum()
```

Reply:
Yeah, I obviously wasn't very clear, because your local example is the problem I was talking about.  I may be confusing terminology somewhere.

Your Series has two data points, `[1, 2]`.

The `len` of the list of labels in your groupby is also two, `['first', 'third']`.

Those two being equal are the problem.

Reply:
OK now actually on a subsequent read I think this is right here